src/prefect/types/__init__.py

```diff
@@ -153,6 +154,10 @@ def convert_none_to_empty_dict(v: Optional[KeyValueLabels]) -> KeyValueLabels:
 ]
 
 
+def uuid7() -> UUID:
+    return cast(UUID, _uuid7())
```

Comment:
`prefect.types` doesn't feel like the right place for this. Since this is mainly a shim for typing, maybe it could go somewhere in `prefect._internal`?